app/lib/frontend/handlers.dart

```diff
@@ -266,16 +268,29 @@ Future<shelf.Response> packagesHandlerHtml(
   shelf.Request request,
   int page, {
   String basePath,
-  String detectedType,
+  PlatformPredicate platformPredicate,
   String title,
   String faviconUrl,
   String descriptionHtml,
 }) async {
   final offset = PackageLinks.RESULTS_PER_PAGE * (page - 1);
   final limit = PackageLinks.MAX_PAGES * PackageLinks.RESULTS_PER_PAGE + 1;
 
-  final packages = await backend.latestPackages(
-      offset: offset, limit: limit, detectedType: detectedType);
+  List<Package> packages;
+  if (platformPredicate != null && platformPredicate.isNotEmpty) {
+    final results = await searchClient.search(new SearchQuery(
+      null,
+      platformPredicate: platformPredicate,
+      order: SearchOrder.updated,
+      offset: offset,
+      limit: limit,
+    ));
+    packages =
+        await backend.lookupPackages(results.packages.map((ps) => ps.package));
+  } else {
+    packages = await backend.latestPackages(offset: offset, limit: limit);
```

Comment:
This leaves the listing of `/packages/` coming from Datastore. We could spare a few lines if this was coming from `search` too, but I left it here, because it has one remote call instead of two.